addons/l10n_my_edi/models/account_edi_xml_ubl_my.py

```diff
@@ -146,6 +150,14 @@ def _export_invoice_vals(self, invoice):
                 },
             })
 
+        vals['vals'].update({
+            'prepaid_payment_vals': {
+                'currency': invoice.currency_id,
+                'currency_dp': self._get_currency_decimal_places(invoice.currency_id),
+                'amount': 0.0 if document_type_code in {'02', '12'} else invoice.amount_total - invoice.amount_residual,
```

Comment:
After discussing it with @kidoodoo we think that you can just always set the amount here.
It will simplify this line, and in the end, the documentation doesn't state that it should be set to 0 for credit notes.